pkg/segment/results/segresults/segresults.go

```diff
@@ -919,3 +906,250 @@ func CreateMeasResultsFromAggResults(limit int,
+	sr.updateLock.Lock()
+	defer sr.updateLock.Unlock()
+
+	remoteStats := &RemoteStats{
+		SegStats: sr.runningSegStat,
+	}
+
+	remoteStats.EvalStats = make(map[string]EvalStatsMetaData)
+	for _, measureAgg := range sr.sAggs.MeasureOperations {
+		if measureAgg.ValueColRequest != nil {
+			switch measureAgg.MeasureFunc {
+			case utils.Min, utils.Max, utils.Count, utils.Sum:
+				_, exist := sr.segStatsResults.measureResults[measureAgg.String()]
+				if exist {
+					remoteStats.EvalStats[measureAgg.String()] = EvalStatsMetaData{
+						MeasureResult: sr.segStatsResults.measureResults[measureAgg.String()].CVal,
+					}
+				}
+			case utils.Range:
+				metadata, exist := sr.runningEvalStats[measureAgg.String()]
+				if exist {
+					_, isRangeStat := metadata.(*structs.RangeStat)
+					if !isRangeStat {
+						return nil, fmt.Errorf("GetRemoteStats: RangeStat not found for range agg %v, qid=%v", measureAgg.String(), sr.qid)
+					}
+					remoteStats.EvalStats[measureAgg.String()] = EvalStatsMetaData{
+						RangeStat: metadata.(*structs.RangeStat),
+					}
+				}
+			case utils.Avg:
+				metadata, exist := sr.runningEvalStats[measureAgg.String()]
+				if exist {
+					_, isAvgStat := metadata.(*structs.AvgStat)
+					if !isAvgStat {
+						return nil, fmt.Errorf("GetRemoteStats: AvgStat not found for avg agg %v, qid=%v", measureAgg.String(), sr.qid)
+					}
+					remoteStats.EvalStats[measureAgg.String()] = EvalStatsMetaData{
+						AvgStat: metadata.(*structs.AvgStat),
+					}
+				}
+			case utils.Cardinality, utils.Values:
+				metadata, exist := sr.runningEvalStats[measureAgg.String()]
+				if exist {
+					_, isStrSet := metadata.(map[string]struct{})
+					if !isStrSet {
+						return nil, fmt.Errorf("GetRemoteStats: String set not found for agg %v, qid=%v", measureAgg.String(), sr.qid)
+					}
+					remoteStats.EvalStats[measureAgg.String()] = EvalStatsMetaData{
+						StrSet: metadata.(map[string]struct{}),
+					}
+				}
+			case utils.List:
+				metadata, exist := sr.runningEvalStats[measureAgg.String()]
+				if exist {
+					_, isStrList := metadata.([]string)
+					if !isStrList {
+						return nil, fmt.Errorf("GetRemoteStats: String list not found for agg %v, qid=%v", measureAgg.String(), sr.qid)
+					}
+					remoteStats.EvalStats[measureAgg.String()] = EvalStatsMetaData{
+						StrList: metadata.([]string),
+					}
+				}
+			}
+		}
+	}
+
+	return remoteStats, nil
+}
+
+func (rs *RemoteStats) RemoteStatsToJSON() (*RemoteStatsJSON, error) {
+	var err error
+
+	remoteStatsJson := &RemoteStatsJSON{
+		EvalStats: rs.EvalStats,
+	}
+
+	remoteStatsJson.SegStats = make([]*structs.SegStatsJSON, len(rs.SegStats))
+	for idx, segStat := range rs.SegStats {
+		if segStat == nil {
+			continue
+		}
+		remoteStatsJson.SegStats[idx], err = segStat.ToJSON()
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return remoteStatsJson, nil
+}
+
+func (rj *RemoteStatsJSON) ToRemoteStats() (*RemoteStats, error) {
```

Comment:
nit: it would be good to have unit tests for `ToRemoteStats` and `RemoteStatsToJSON`. A simple test where you start with one, convert to the other, and then convert back and verify you get what you started with would be good

Reply:
addressed in the latest commit.